packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -417,7 +417,14 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
       // Normally we can bail out on props equality but if context has changed
       // we don't do the bailout and we have to reuse existing props instead.
     } else if (memoizedProps === nextProps) {
-      return bailoutOnAlreadyFinishedWork(current, workInProgress);
+      if (
+        workInProgress.mode & AsyncMode &&
+        shouldDeprioritizeSubtree(type, nextProps)
+      ) {
+        // Don't bail-out if this is hidden
```

Comment:
As a general style guide we should invert this logic and remove the dead block because it is not always possible for the compiler to invert these conditionals and strip the extra code. Due to things like `!(NaN > NaN)` isn't the same as `NaN <= NaN`

Reply:
I mean it is always possible to wrap in `!()` but those three extra bytes can't be reclaimed.